src/index.js

```diff
@@ -8,26 +8,31 @@ const options = Object.freeze({
 
 const NightModeToggle = ({ size, checked, onChange, ...extraProps }) => {
   const ref = useRef();
-  const [progress, setProgress] = useState(() => 0);
+  const [progress, setProgress] = useReducer((_,next) => next, 0);
+
   useEffect(() => {
     if (progress >= 0.5) {
-      if (checked) {
+      if (ref.current && checked) {
```

Comment:
My understanding of `ref` is that once the DOM is mounted, we'll have access to `ref.current`. But i agree, there's no harm in doing this and makes potential issues in future easier to debug.